src/backend/src/services/description-bullets.services.ts

```diff
@@ -0,0 +1,57 @@
+import { User, WBS_Element_Status } from '@prisma/client';
+import prisma from '../prisma/prisma';
+import { hasBulletCheckingPermissions } from '../utils/description-bullets.utils';
+import { AccessDeniedException, HttpException, NotFoundException } from '../utils/errors.utils';
+import { descBulletTransformer } from '../transformers/description-bullets.transformer';
+import { descBulletArgs } from '../prisma-query-args/description-bullets.query-args';
+
+export default class DBService {
+  static async checkDescriptionBullet(userId: number, descriptionId: number, user: User) {
+    const originalDB = await prisma.description_Bullet.findUnique({
+      where: { descriptionId },
+      include: {
+        workPackageDeliverables: { include: { wbsElement: true } },
+        workPackageExpectedActivities: { include: { wbsElement: true } }
+      }
+    });
+    if (!originalDB) throw new NotFoundException('Description Bullet', descriptionId);
+
+    if (originalDB.dateDeleted) throw new HttpException(400, 'Cant edit a deleted Description Bullet!');
+
+    const workPackage = originalDB.workPackageDeliverables || originalDB.workPackageExpectedActivities;
+
+    if (!workPackage)
+      throw new HttpException(400, 'This description bullet is not tied to a workpackage deliverable or expected activity');
+
+    if (workPackage.wbsElement.status !== WBS_Element_Status.ACTIVE)
+      throw new HttpException(400, 'Cannot check a description bullet on an inactive work package!');
+
+    const hasPerms = await hasBulletCheckingPermissions(userId, descriptionId);
```

Comment:
this should be checked usign user.userId

Reply:
might have made the change? not sure